R/loadReg2.R

```diff
@@ -323,42 +324,66 @@ predictSolute.loadReg2 <- function(
   # default. nchunks is the number of chunks required.
   chunk.size <- 176000
   nchunks <- ceiling(nrow(newdata) / chunk.size)
-  newdata <- lapply(1:nchunks, function(i) {
+  datachunks <- lapply(1:nchunks, function(i) {
     newdata[((i-1)*chunk.size + 1):min(i*chunk.size, nrow(newdata)),]
   })
   
   # Now do the prediction for each chunk and then reassemble the full set of
   # predictions
-  preds_raw <- lapply(newdata, function(datachunk) {
+  preds_lin_raw <- lapply(datachunks, function(datachunk) {
     switch(
       flux.or.conc,
-      "flux"=predLoad(
+      "flux"={
         # ... may contain load.units, seopt, print
-        fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level, ...), 
-      "conc"=predConc(
-        fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level) 
+        predLoad(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level, ...)
+      }, 
+      "conc"={
+        predLoad_args <- c('load.units','seopt','print')[(c('load.units','seopt','print') %in% names(list(...)))]
+        if(length(predLoad_args) > 0) warning(paste("these args are ignored for flux.or.conc='conc':", paste(predLoad_args, collapse=', ')))
+        predConc(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level) 
+      }
     )
   })
-  preds_raw <- do.call(rbind, preds_raw)
+  preds_lin_raw <- do.call(rbind, preds_lin_raw)
+  preds_col <- .sentenceCase(flux.or.conc)
```

Comment:
me too! i did just find this:
```r
tools::toTitleCase('one by one')
```
but think i'll stick to `loadflex:::.sentenceCase` here since it's already written.